ChatSpace チャット画面実装

YamamotoMao35 opened the pull request:
# WHAT
ChatSpaceのチャット画面を実装する。

# WHY
ChatSpaceのチャット画面実装は、追加機能を搭載するための基礎となるため。
チャット画面から次にどのような機能が必要かわかりやすくなる。


![](https://i.gyazo.com/00718ea838433d9185959e32b2f3fc32.png)

Commit:
Add Gem

Add Gem

Gemfileに「haml-rails」を入れてhaml記法ができるようにし、「erb2haml」を入れることでerbファイルの変換ができるようになった。

Commit:
Convert .erb into haml

Convert .erb into haml

.erbファイルを.hamlファイルに変換した。

Commit:
Create messages_controller

Create messages_controller

messagesコントローラーを作成し、そのindexアクションに対応するビューとルートパスも作成した。

Commit:
Modify file name

Modify file name

間違えて.erbとなっていたため.hamlに修正

Commit:
Change filename .css

Change filename .css

application.cssの拡張子を.scssにして、リセットCSSの読み込みを定義した

Commit:
Create _reset.scss

Create _reset.scss

リセットCSSを作成した。

Commit:
Create Outer view

Create Outer view

「flash-box」というクラスでビューの外枠を作成した。

Commit:
Update index.html.haml

Update index.html.haml

ページ画面のサイドに使うクラスを作成した。

Commit:
Create side-header

Create side-header

サイドバーのヘッダー部を作成した。

Commit:
Create side-body

Create side-body

サイドバーのヘッダー以下およびグループ名およびメッセージが表示される欄を作成した。

Commit:
Add gem "font-awesome-rails"

Add gem "font-awesome-rails"

gem「font-awesome-rails」を導入した。

Commit:
Mount view-side icon

Mount view-side icon

「font-awesome-rails」のアイコンをサイドバーに実装した。

Commit:
Create Chat-main View

Create Chat-main View

ChatSpaceのチャットを打つ画面の枠組みを作成した。

Commit:
Update index.html.haml

Update index.html.haml

メイン画面のヘッダーのグループと名前が表示される欄を作成した。

Commit:
Create Edit-btn

Create Edit-btn

Editボタンを作成した。

Commit:
Create log-in notification

Create log-in notification

ログイン時やグループ作成時に出る通知を表示する枠を作成した。

Commit:
Create Chat-message

Create Chat-message

メイン画面内にチャットが表示できるようにした。

Commit:
Create text input box

Create text input box

チャットを打ち込む箱を作成した。

Commit:
Create Send-btn

Create send-btn

メイン画面のチャット入力欄に「Send」ボタンを設置した。
またチャット入力欄に何も入力されてないときは「type a message」と表示するようにした。

Commit:
Mount image-icon

Mount image-icon

テキスト入力欄にアイコンを設置し、画像を取り込めるようにした

Commit:
Create Partial template

Create Partial template

部分テンプレートを作成した。

Commit:
Arrange files

Arrange files

このコミットで行うこと
・_group.scssと_main.scssをそれぞれside-viewフォルダ、main-viewフォルダに振り分けた。この2つのフォルダはstylesheetの下に作成した。またside-viewフォルダには_side.scssを作成した。このファイルの中にサイドバーのビュー枠の設定を入れた。

・hamlファイル内のaタグをlink_toに変更した。aタグは基本的にlink_toメソッドに置き換えるため。

・変数名を変更して色についての変数だとわかるようにした。

・_style.scss内の不要な記述を削除した。

・各ファイルのインデント調整

TOMARUMARU commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,10 @@
+!!!
+%html
+  %head
+    %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
ハッシュロケット修正しましょう

YamamotoMao35 replied:
修正しました。

TOMARUMARU commented on app/views/layouts/mailer.html.haml:
```diff
@@ -0,0 +1,8 @@
+    :css
+      /* Email styles need to be inline */
+  %body
+    = yield
```
mailer系のファイルは使用しないので消しちゃって大丈夫です

YamamotoMao35 replied:
修正しました

TOMARUMARU commented on app/views/layouts/mailer.text.haml:
```diff
@@ -0,0 +1 @@
+= yield
```
こちらも

YamamotoMao35 replied:
修正しました

TOMARUMARU commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,38 @@
+!!!
+%html
+%head
+  %body
```
ここから上の記述は不要です

YamamotoMao35 replied:
修正しました。

TOMARUMARU commented on test/controllers/messages_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
使用しないファイルは消しましょう

YamamotoMao35 replied:
修正しました。

TOMARUMARU reviewed (commented):
ファイルの最後は改行を１つ入れましょう

Commit:
Delete mailer file

Delete mailer file

mailer系のファイルは使用しないため削除した。

Commit:
Delete test.rb

Delete test.rb

アプリケーションをテストするためのファイルは今回不要であるため削除。またその原因になっていたapplication.rbの誤記も修正

Commit:
Modify files

Modify files

このコミットで行うこと
・ファイルの最後に改行が入るようにした。
・application.html.hamlのハッシュロケットを修正した。
・index.html.hamlの1〜4行目にあった記述を削除し、インデントを詰めた。

ta910 commented:
LGTMです！
今後、PRを出す時はWIPを外して出すようにしましょう
